install/resources/upgrade34.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * MyBB 1.8
+ * Copyright 2014 MyBB Group, All Rights Reserved
+ *
+ * Website: http://www.mybb.com
+ * License: http://www.mybb.com/about/license
+ *
+ */
+
+/**
+ * Upgrade Script: 1.8.5
+ */
+
+$upgrade_detail = array(
+	"revert_all_templates" => 0,
+	"revert_all_themes" => 0,
+	"revert_all_settings" => 0
+);
+
+@set_time_limit(0);
+
+function upgrade34_dbchanges()
+{
+	global $db, $output;
+
+	$query = $db->simple_select('settings', 'value', "name='username_method'");
```

Comment:
The `$settings` array is still build from a cache file which content may differ from that of the DB.

Just because somebody else opted to ignore the slight possibilities of that being the case doesn't mean I will do the same.

Please provide realistic reasons as to why I should take the time to modify something that I considered to be the most effective approach other that just one single meaningless query resources. Which I already thought about before pushing my changes.

Now if somebody else supports you I will not resist.

BTW those references only make use of the setting value not actually update values as this case does.